### PR TITLE
native payment multichains

### PR DESCRIPTION
**Pull request by benjlevesque**
- fix: Celo token names
- lower case tokens
- introduce native tokens
- feat: Native tokens, cUSD case fix, misc currency refactoring
- revert unnecessary change
- RIN + native changes
- feat: native token payment for evm chains

### CODE REVIEW

File: packages/currency/test/currency.test.ts [hunk 1]
```diff
@@ -213,9 +296,47 @@ describe('api/currency', () => {
         value: 'EUR',
       });
     });
+
+    describe('errors and edge cases', () => {
```
Comment (benjlevesque) [inline packages/currency/test/currency.test.ts:300]: moved code only

File: packages/payment-detection/src/eth/multichainExplorerApiProvider.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  sokol: { chainId: 77, name: 'sokol' },
+  fuse: { chainId: 122, name: 'fuse' },
+  celo: { chainId: 42220, name: 'celo' },
+  fantom: { chainId: 250, name: 'fantom' },
+};
+
+/**
+ * A provider that supports multiple APIs, like Etherscan and Blockscout
+ */
+export class MultichainExplorerApiProvider extends ethers.providers.EtherscanProvider {
```
Comment (yomarion) [inline packages/payment-detection/src/eth/multichainExplorerApiProvider.ts:13]: Smart move !

**Review by yomarion — APPROVED** [review]
Solid and elegant !

**Review by yomarion — APPROVED** [review]
